homeassistant/components/zha/entities/entity.py

```diff
@@ -103,3 +103,16 @@ def device_info(self):
             'name': self._device_state_attributes['friendly_name'],
             'via_hub': (DOMAIN, self.hass.data[DATA_ZHA][DATA_ZHA_BRIDGE_ID]),
         }
+
+    @callback
+    def zha_send_event(self, cluster, command, args):
+        """Relay entity events to hass."""
+        self.hass.bus.async_fire(
```

Comment:
Entities shouldn't fire custom events. They already spawn state changed events. Devices that need to fire events should not be entities and the events should be fired from the zha component.

Reply:
See `DeconzEvent` in the deconz package for an example.

Reply:
I had it like that at first... may I ask why this is the case? I can do that but all it does is artificially increase the number of objects being managed. These entities are going to exist anyway and there will be a platform level event object for each one as well. I thought this was a much cleaner implementation.

Reply:
How do you mean platform level event?

Entities provide states, and should not fire other types of events.

If entities are not providing states, they should not be entities. What's the case here?

Reply:
I reworked this. Let me know what you think.

Reply:
changes pushed